### PR TITLE
Standard code formatting

### PR DESCRIPTION
**Pull request by chick**
We have talked about the need for consistent code formatting for a while now. Consistent formatting
should  make our code easier to read. Automating the formatting ensures that the formatting is consistent no matter who works on the code. This is more important as we have new contributors with different personal styles.

**Resources**:
[https://scalameta.org/scalafmt/](https://scalameta.org/scalafmt/)

**Type of change**: Code formatting

**Impact**: No functional change.
Visually different code

**Development Phase**: Proposal
I would encourage interested parties to play with the settings in the file. I personally like the more
open style but it comes in some cost in total number of lines

If adopted we should make a decision about doing a pass over every file or perhaps at first 
reformatting files as they are changed in other PRs

**Release Notes**
Chisel3 is moving to a standard automated code formatting style

### CODE REVIEW

**Comment by ducky64** [comment]
Seems like a good idea. Where'd your .scalafmt.config come from? It seems to have a mix of defaults (so redundant lines?) and custom overrides (like `continuationIndent.defnSite = 2` which I disagree with).

Looking through the example code change, a major discussion point will probably be whether we want the vertical alignment (which is somewhat enabled as default).

There also seem to be a few cases where it does things inconsistently - for example, in a block of match-case statements, some of the cases have newlines and some don't.

**Comment by chick** [comment]
@ducky64 I have now started with the .scalafmt.config fromt the scalafmt configuration webpage. It should make discovery or fiddling around with it a bit easier. The only think I changed was the columns to be 100. There do appear to be some inconsistencies, but I would guess there are similar inconsistencies in most developers hand formatted code

**Comment by seldridge** [comment]
Are there any thoughts on how to enforce this going forward? 

For IntelliJ users, things should "just work" as long as it's configured correctly. As a catch-all, does a pre-commit hook make sense? 

A CI check would work, too, though that should ideally be one of the first things that runs to keep the feedback fast.

File: .scalafmt.config [hunk 1]
```diff
@@ -0,0 +1,164 @@
+version = "2.0.0"
+maxColumn = 100
```
Comment (seldridge) [inline .scalafmt.config]:
```suggestion
maxColumn = 120
```

**Comment by chick** [comment]
Will create new PR for this later when PR queue is smaller

**Pull request closed without merging**